webapp/channels/src/plugins/registry.ts

```diff
@@ -279,7 +306,7 @@ export default class PluginRegistry {
     }: {
         button: ReactResolvable;
         dropdownButton: ReactResolvable;
```

Comment:
For certain elements (mainly ReactResolvable), what the registry receives is not exactly what the component will receive. You see that what the registry receive, we pass it through the `resolveReactElement`.

This is done so you can pass a element by name string (not sure why would you do that, but it is supported...). If we were to put `CallButtonAction['dropdownButton']`, in theory, it may not allow that possibility.

The reality though is that a string is still a ReactNode, so it should be fine, but this definition is more "correct" and "intention revealing".

Reply:
Thanks for the thorough explanation, appreciate it!